### PR TITLE
Make Samba and dependencies respect a python USE flag

### PR DESCRIPTION
**Pull request by illiminable**
This adds support for python USE flag to ldb and Samba and makes a minor cleanup in talloc and tevent.

If we are happy with these changes I can apply them to the earlier ebuilds where applicable in the various projects too.

I've also been talking to jra@samba about getting all these python patches upstream to the Samba repository. Discussion here [1]. I'm happy to do some of the legwork if necessary but they do require the original author polynomial-c@ to sign off before they can accept them. See [2]

[1] - https://lists.samba.org/archive/samba-technical/2017-January/117923.html
[2] - https://www.samba.org/samba/devel/copyright-policy.html

### CODE REVIEW

**Comment by Polynomial-C** [comment]
The original author of these python patches is @axs-gentoo. I only forward-ported a couple of patches once the original patches from @axs-gentoo didn't apply anymore.

**Comment by axs-gentoo** [comment]
I think the authorship <-> samba upstream needs discussion should likely be held off-github. :)  let's switch to email and I'll see about commenting on the samba thread.

**Comment by Polynomial-C** [comment]
Added the "do not merge" label until the issues I've found have been fixed.

**Comment by illiminable** [comment]
Is there something on this change you are waiting on from me? Or should I go ahead and make these changes to other relevant ebuild versions?

**Comment by Polynomial-C** [comment]
Right now the PR would make all touched packages unuseable as we need to run `python-single-r1_pkg_setup` unconditionally (WAF is a python build system).
See my various comments on the corresponding changes.

**Comment by illiminable** [comment]
I see - based on the doc I thought use python && was the right thing to do but I guess I didn't understand.

I will remove those parts. 

BTW - I don't see any of your review comments.

**Comment by Polynomial-C** [comment]
Sorry, it seems I forgot to submit the change requests (I did this for the first time). Now they should be visible.

File: net-fs/samba/samba-4.5.3.ebuild [hunk 1]
```diff
@@ -40,7 +40,8 @@ MULTILIB_WRAPPED_HEADERS=(
 )
 
 # sys-apps/attr is an automagic dependency (see bug #489748)
-CDEPEND="${PYTHON_DEPS}
+CDEPEND="
+	python? ( ${PYTHON_DEPS} )
```
Comment (mgorny) [inline net-fs/samba/samba-4.5.3.ebuild]: You also need PYTHON_DEPS in DEPEND because of waf.

File: net-fs/samba/samba-4.5.3.ebuild [hunk 2]
```diff
@@ -81,7 +82,7 @@ RDEPEND="${CDEPEND}
 
 REQUIRED_USE="addc? ( gnutls !system-mitkrb5 )
 	ads? ( acl gnutls ldap )
-	${PYTHON_REQUIRED_USE}"
+	python? ( ${PYTHON_REQUIRED_USE} )"
```
Comment (mgorny) [inline net-fs/samba/samba-4.5.3.ebuild]: This needs to be unconditional because of waf.

**Comment by mgorny** [comment]
Ping.

**Comment by monsieurp** [comment]
Hi @illiminable. Please let us know whether you're still interested in working on this PR. Let's close it for now.

**Pull request closed without merging**